### PR TITLE
Use multiple classes dirs in Classycle plugin

### PR DESCRIPTION
**Pull request by lptr**
This is to upgrade Classycle used in the Gradle build to use Gradle 4.0-compatible way to access compiled classes produced from source sets.

### CODE REVIEW

**Comment by lptr** [comment]
We also used to print a huge stacktrace before, that's fixed now:

```text
groovy.lang.MissingPropertyException: No such property: rootProject for class: org.apache.tools.ant.Project
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:53)
        at org.codehaus.groovy.runtime.callsite.GetEffectivePojoPropertySite.getProperty(GetEffectivePojoPropertySite.java:66)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:296)
        at org.gradle.plugins.classycle.Classycle$_generate_closure1.doCall(Classycle.groovy:86)
```

File: buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy [hunk 1]
```diff
@@ -42,7 +42,7 @@ class ClassyclePlugin implements Plugin<Project> {
             def taskName = sourceSet.getTaskName('classycle', null)
             project.tasks.create(taskName, Classycle, { Classycle task ->
                 task.reportName = sourceSet.name
-                task.classesDir = sourceSet.output.classesDir
+                task.classesDirs = sourceSet.output.classesDirs.files.findAll { it.exists() }
```
Comment (wolfs) [inline buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy]: Don't we resolve the file collection here too early? This now happens at configuration time before any of the `classesDirs` exist, right?
I think this should happen during execution when iterating over the directories.

Comment (lptr) [inline buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy]: I tested this and it worked, but I think you have a point. Updated.

Comment (wolfs) [inline buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy]: Did you run a clean build first?

File: buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy [hunk 2]
```diff
@@ -42,7 +42,7 @@ class ClassyclePlugin implements Plugin<Project> {
             def taskName = sourceSet.getTaskName('classycle', null)
             project.tasks.create(taskName, Classycle, { Classycle task ->
                 task.reportName = sourceSet.name
-                task.classesDir = sourceSet.output.classesDir
+                task.classesDirs = project.files { sourceSet.output.classesDirs.files.findAll { it.exists() } }
```
Comment (wolfs) [inline buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy]: As @oehme stated, this should happen when the task executes. I don't even know why we need the empty directory handling at all. But if we want to do it, it should happen when the task executes in its task action.

Comment (lptr) [inline buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy]: Fixed.

**Pull request merged**